app/models/hotel.rb

```diff
@@ -1,3 +1,10 @@
 class Hotel < ApplicationRecord
   has_many :rooms
+  has_many :guests, through: :rooms
+
+  def guest_list
+    # I also googled this eleventeen thousand times to ensure it's active record but
+    # TBD...
+    guests.distinct
+  end
```

Comment:
Hah yes... it is indeed an AR method ;) Nice work. And, good job pulling this into a model method in your Hotel class. Since it's data logic, this is exactly where it should go.